src/commands/check.ts

```diff
@@ -0,0 +1,149 @@
+    ...Command.flags,
+    //Inputs
+    profileId: oclifFlags.string({
+      description: 'Profile Id in format [scope](optional)/[name]',
+      required: true,
+    }),
+    providerName: oclifFlags.string({
+      description: 'Name of provider.',
+      required: true,
+    }),
+    scan: oclifFlags.integer({
+      char: 's',
+      description:
+        'When number provided, scan for super.json outside cwd within range represented by this number.',
+      required: false,
+    }),
+    json: oclifFlags.boolean({
+      char: 'j',
+      description: 'Formats result to JSON',
+    }),
+  };
+
+  static examples = ['$ station check', '$ station check -q'];
+
+  private logCallback? = (message: string) => this.log(grey(message));
+  private warnCallback? = (message: string) => this.log(yellow(message));
+
+  async run(): Promise<void> {
+    const { flags } = this.parse(Check);
+
+    if (flags.quiet) {
+      this.logCallback = undefined;
+      this.warnCallback = undefined;
+    }
+
+    if (flags.scan && (typeof flags.scan !== 'number' || flags.scan > 5)) {
+      throw userError(
+        '--scan/-s : Number of levels to scan cannot be higher than 5',
+        1
+      );
+    }
+
+    //Load super json
+    const superPath = await detectSuperJson(process.cwd(), flags.scan);
+    if (!superPath) {
+      throw userError('Unable to check, super.json not found', 1);
+    }
+    const loadedResult = await SuperJson.load(joinPath(superPath, META_FILE));
+    const superJson = loadedResult.match(
+      v => v,
+      err => {
+        throw userError(`Unable to load super.json: ${err}`, 1);
+      }
+    );
+
+    const parsedProfileId = parseDocumentId(flags.profileId);
+    if (parsedProfileId.kind == 'error') {
+      throw userError(`Invalid profile id: ${parsedProfileId.message}`, 1);
+    }
+
+    if (!isValidProviderName(flags.providerName)) {
+      throw userError(`Invalid provider name: "${flags.providerName}"`, 1);
+    }
+
+    const profile: {
+      name: string;
+      scope?: string;
+      version?: string;
+    } = {
+      name: parsedProfileId.value.middle[0],
+      scope: parsedProfileId.value.scope,
+    };
+
+    //Get profile info
+    const profileSettings = superJson.normalized.profiles[flags.profileId];
+    if (!profileSettings) {
+      throw userError(
+        `Profile id: "${flags.profileId}" not found in super.json`,
+        1
+      );
+    }
+
+    if ('version' in profileSettings) {
+      profile.version = profileSettings.version;
+    }
+
+    //Get map info
+    const map: {
+      variant?: string;
+    } = {};
+    const profileProviderSettings =
+      superJson.normalized.profiles[flags.profileId].providers[
+        flags.providerName
+      ];
+
+    if (!profileProviderSettings) {
+      throw userError(
+        `Provider: "${flags.providerName}" not found in profile: "${flags.profileId}" in super.json`,
+        1
+      );
+    }
+
+    //TODO: how to resolve map revision??
```

Comment:
Also from super.json? Or what's the problem?

Reply:
I am just not sure if we use map revision in fetching map AST or somewhere else 🤷